packages/vite/src/node/plugins/asset.ts

```diff
@@ -187,10 +188,13 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
     generateBundle(_, bundle) {
       // do not emit assets for SSR build
       if (config.command === 'build' && config.build.ssr) {
-        for (const file in bundle) {
+        const assetFiles = Object.keys(bundle).filter(
+          (file) => bundle[file].type === 'asset',
+        )
+        for (const file of assetFiles) {
           if (
-            bundle[file].type === 'asset' &&
-            !file.includes('ssr-manifest.json')
+            !file.endsWith('ssr-manifest.json') &&
+            !jsSourceMapRE.test(file)
```

Comment:
I think we should avoid the `Object.keys` and `filter` here as they create new objects that takes up memory. We could stick with the for in loop like before and add the sourcemap check for a tiny bit of perf.

Reply:
I will make that change, I've been OOO on vacation the past few weeks